python/openchange/migration/directory.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# OpenChangeDB DB schema and its migrations
+# Copyright (C) Enrique J. Hernández Blasco <ejhernandez@zentyal.com> 2015
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Migration for OpenChange directory schema and data
+"""
+from openchange.migration import migration, Migration
+import ldb
+
+from samba.samdb import SamDB
+from samba.auth import system_session
+from samba.provision import setup_modify_ldif
+from samba.net import Net
+from samba.dcerpc import nbt
+
+# this helper functions are extracted from openchange.provision to avoid circular import
+# it would be better a refactoring of provision.ppy so we can avoid repeating code here
+
```

Comment:
Since those short methods and unlikely to change, I will let them her for the moment. Later, if we have another py file which use them we can move them